src/hotspot/share/prims/jvmtiEventController.cpp

```diff
@@ -824,6 +824,14 @@ JvmtiEventControllerPrivate::set_extension_event_callback(JvmtiEnvBase *env,
     case EXT_EVENT_VIRTUAL_THREAD_UNMOUNT :
       ext_callbacks->VirtualThreadUnmount = callback;
       break;
+    case EXT_EVENT_CRAC_BEFORE_CHECKPOINT :
+      ext_callbacks->CracBeforeCheckpoint = callback;
+      env->env_event_enable()->set_user_enabled(event_type, enabling);
```

Comment:
Why UNLOAD/MOUNT/UNMOUNT do not require this?

Reply:
For EXT_EVENT_CLASS_UNLOAD, it was here from the very beginning (https://github.com/openjdk/crac/blob/8153779ad32d1e8ddd37ced826c76c7aafc61894/hotspot/src/share/vm/prims/jvmtiEventController.cpp#L693), until VitualThreads implementation changed it. 
Now a consumer (an agent) should set a callback for an event, and then can control recieving an event by enabling/disabling it with a call of `SetEventNotificationMode ` function. I guess, this was done to make the event control flow easier for a consumer. This makes sense in case an event is supposed to posted regularly, that is not a case for CRaC. Again, we could implement a similar way, but it still seems like one more place for conflicts in future merging.